### ISSUE TITLE
New HEEx syntax in 2.4.0 breaks templates for pre-v1.0 versions of LiveView

### ISSUE THREAD

**Issue by arcanemachine** [team-alembic/ash_authentication_phoenix#547]
TLDR: For now, to support pre-v1.0 versions of LiveView, use the following Igniter task when setting up AshAuthenticationPhoenix:

`mix igniter.install ash_authentication_phoenix@2.3.0`

---

I used Igniter task to install AshAuthentication to an existing Ash project which was using LiveView version `1.0.0-rc.1`:

`mix igniter.install ash_authentication`

I then added the password auth strategy and ran the migrations:

```
mix ash_authentication.add_strategy password
mix ash.migrate
```

I then ran the Igniter task to set up AshAuthenticationPhoenix:

`mix igniter.install ash_authentication_phoenix`

I then added the snippet to `assets/tailwind.config.js`, as mentioned in the terminal after running the Igniter task.

---

After all this, I restarted the server. However, when navigating to any of the auth routes in the browser, the browser only renders the following text:

`{@inner_content}`

---

Downgrading AshAuthenticationPhoenix to `2.3.0` fixes the issue.

Is this something that could be handled by an Igniter task that detects which version of LiveView is installed (e.g. for LiveView `0.20.x` and the `1.0-rc.x` versions, install AshAuthenticationPhoenix `2.3.0`, but install `2.4.0` for LiveView `1.0` or greater)?

Thanks

**Comment by zachdaniel** [comment]
🤔 I would have expected hex version resolution to have solved this. `2.4` says that it requires liveview `~> 1.0`. So it seems strange that you would have been able to have LV `1.0.0-rc.1` in the same application as `AshAuthenticationPhoenix` 2.4.0.

**Comment by zachdaniel** [comment]
Do you have any `override: true` in your dependencies by chance?

**Comment by arcanemachine** [comment]
Just one: The project was auto-generated with the following LiveView version:

`{:phoenix_live_view, "~> 1.0.0-rc.1", override: true},`

**Comment by zachdaniel** [comment]
I think that might be coming from `mix phx.new`? I don't think anything igniter does would set an override on the phoenix version. If that is the case, then there isn't really anything we can do about it unfortunately. We rely on mix version resolution for compatibility. If we had to check it all manually it would be pretty error prone I think.

This is a useful issue for others that run into this issue or any similarly strange issues, but ultimately the problem should go away w/ the latest version of phx.new not using an override (I hope?)

**Comment by arcanemachine** [comment]
No, the newest Phoenix generator does not have any overrides.

For now, the TLDR at the top of my first post should provide a reasonable workaround for users who cannot upgrade.
